File: frontend/src/services/expense.js
import { getDB, saveDB, nextId, delay, getCurrentUser } from './mockData';

export const expenseService = {
    addExpense: async (expenseData) => {
        await delay();
        const db = getDB();
        const user = getCurrentUser();
        const roomMembers = db.users.filter(u => u.room_id === user.room_id);
        const splitAmount = Math.round(expenseData.amount / roomMembers.length);

        const expense = {
            id: nextId(db.expenses),
            room_id: user.room_id,
            title: expenseData.title,
            amount: expenseData.amount,
            category: expenseData.category || 'General',
            paid_by: user.id,
            paid_by_name: user.name,
            created_at: new Date().toISOString(),
            splits: roomMembers.map((m, i) => ({
                id: Date.now() + i,
                user_id: m.id,
                user_name: m.name,
                amount: splitAmount,
                is_paid: m.id === user.id,
            })),
        };
        db.expenses.push(expense);

        // Add notification for other members
        roomMembers.filter(m => m.id !== user.id).forEach(m => {
            db.notifications.push({
                id: nextId(db.notifications),
                user_id: m.id,
                title: 'New Expense Added',
                message: `${user.name} added ${expenseData.title} - ₹${expenseData.amount.toLocaleString()}`,
                type: 'expense',
                is_read: false,
                created_at: new Date().toISOString(),
            });
        });

        saveDB(db);
        return expense;
    },

    getRoomExpenses: async (roomId) => {
        await delay();
        const db = getDB();
        return db.expenses.filter(e => e.room_id === roomId);
    },

    settleSplit: async (splitId) => {
        await delay();
        const db = getDB();
        for (const expense of db.expenses) {
            const split = expense.splits?.find(s => s.id === splitId);
            if (split) {
                split.is_paid = true;
                break;
            }
        }
        saveDB(db);
        return { message: 'Split settled' };
    },
};

File: backend/app/schemas/expense.py
from typing import Optional, List
from pydantic import BaseModel
from datetime import datetime

class ExpenseSplitBase(BaseModel):
    user_id: int
    amount_owed: float

class ExpenseSplitCreate(ExpenseSplitBase):
    pass

class ExpenseSplit(ExpenseSplitBase):
    id: int
    expense_id: int
    is_paid: bool

    class Config:
        from_attributes = True

class ExpenseBase(BaseModel):
    title: str
    amount: float
    category: Optional[str] = None
    attachment: Optional[str] = None

class ExpenseCreate(ExpenseBase):
    room_id: int
    splits: List[ExpenseSplitCreate]

class Expense(ExpenseBase):
    id: int
    room_id: int
    paid_by: int
    paid_by_name: Optional[str] = None
    created_at: datetime
    splits: List[ExpenseSplit] = []

    class Config:
        from_attributes = True
